Update how-to-route-public-folder-hierarchy-requests.md (#306)
MicrosoftDocs/office-developer-exchange-docs: docs/exchange-web-services/how-to-route-public-folder-hierarchy-requests.md

|**X-PublicFolderMailbox** <br/> |The [InternalRpcClientServer](/exchange/client-developer/web-service-reference/setting-soap) value from a [GetUserSettings](/exchange/client-developer/web-service-reference/getusersettings-operation-soap) Autodiscover SOAP response, which becomes the value of the **X-PublicFolderMailbox** header.|1. Send a **GetUserSetting** request with the SMTP address for the user's mailbox.<br/><br/>2. Use the **InternalRpcClientServer** element returned by the Autodiscover service to populate the value of the **X-PublicFolderMailbox** header. The value of the **X-PublicFolderMailbox** element is an SMTP address where the user part of the address is a GUID.|

After you have determined the header values, include them [when you make public folder hierarchy requests](#bk_setheadervalues).
  
The steps in this article are specific to public folder hierarchy requests. To determine whether your request is a public folder hierarchy or content request, see [Routing public folder requests](public-folder-access-with-ews-in-exchange.md#bk_routing). 
  

> ## Determine the header values by using the EWS Managed API

Reviewer: ```suggestion
## Determine the header values using the EWS Managed API
```

<a name="bk_getpfinfoewsma"> </a>

> You can use a single call to [GetUserSettings](/exchange/client-developer/web-service-reference/getusersettings-operation-soap) by using the following code, which retrieves both **InternalRpcClientServer** and **PublicFolderInformation** elements. Include the SMTP address of the mailbox user as an input parameter.

Reviewer: ```suggestion
You can use a single call to [GetUserSettings](/exchange/client-developer/web-service-reference/getusersettings-operation-soap) using the following code, which retrieves both **InternalRpcClientServer** and **PublicFolderInformation** elements. Include the SMTP address of the mailbox user as an input parameter.
```

Console.WriteLine("X-AnchorMailbox value for public folder hierarchy requests: {0}", userResponse.Settings[UserSettingName.PublicFolderInformation]);
Console.WriteLine("X-PublicFolderMailbox value for public folder hierarchy requests: {0}", userResponse.Settings[UserSettingName.InternalRpcClientServer]);
```

After running the code, the following information is displayed on the console:
  
`X-AnchorMailbox for public folder hierarchy requests: SharedPublicFolder@contoso.com`<br/>
`X-PublicFolderMailbox value for public folder hierarchy requests: 1ec2a236-ed93-4f88-b9c6-33e63fa4aa44@contoso.com`

Now that you have the **PublicFolderInformation** value, include it as the value for the X-AnchorMailbox header in all public folder hierarchy requests. The **InternalRpcClientServer** value is used for the X-PublicFolderMailbox header.  Both headers are required.
  
`X-AnchorMailbox: SharedPublicFolder@contoso.com`<br/>
`X-PublicFolderMailbox: 1ec2a236-ed93-4f88-b9c6-33e63fa4aa44@contoso.com`

## Determine the the header values using SOAP
<a name="bk_getpfinfoews"></a>

> The following code example shows how to retrieve the **PublicFolderInformation** and **InternalRpcClientServer** values by using the [GetUserSettings](/exchange/client-developer/web-service-reference/getusersettings-operation-soap) SOAP operation. The mailbox user is specified in the [Mailbox](/exchange/client-developer/web-service-reference/mailbox-soap) element, and the [RequestedSettings](/exchange/client-developer/web-service-reference/requestedsettings-soap) element limits the response to the [PublicFolderInformation](/dotnet/api/microsoft.exchange.webservices.autodiscover.usersettingname) and [InternalRpcClientServer](/exchange/client-developer/web-service-reference/setting-soap) values.

Reviewer: ```suggestion
The following code example shows how to retrieve the **PublicFolderInformation** and **InternalRpcClientServer** values using the [GetUserSettings](/exchange/client-developer/web-service-reference/getusersettings-operation-soap) SOAP operation. The mailbox user is specified in the [Mailbox](/exchange/client-developer/web-service-reference/mailbox-soap) element, and the [RequestedSettings](/exchange/client-developer/web-service-reference/requestedsettings-soap) element limits the response to the [PublicFolderInformation](/dotnet/api/microsoft.exchange.webservices.autodiscover.usersettingname) and [InternalRpcClientServer](/exchange/client-developer/web-service-reference/setting-soap) values.
```

The response includes the **PublicFolderInformation** value. 
  
```XML
<UserSetting i:type="StringSetting">
  <Name>InternalRpcClientServer</Name>
  <Value>1ec2a236-ed93-4f88-b9c6-33e63fa4aa44@contoso.com</Value>
</UserSetting>
<UserSetting i:type="StringSetting">
    <Name>PublicFolderInformation</Name>
    <Value>SharedPublicFolder@contoso.com</Value>
</UserSetting>
```

> Now that you have the **PublicFolderInformation** value, include it as the value for the X-AnchorMailbox header in all public folder hierarchy requests.  The **InternalRpcClientServer** value is used for the X-PublicFolderMailbox header.  Both headers are required.

Reviewer: ```suggestion
Now that you have the **PublicFolderInformation** value, include it as the value for the X-AnchorMailbox header in all public folder hierarchy requests. Use the **InternalRpcClientServer** value for the X-PublicFolderMailbox header. Both headers are required.
```
